src/app/pages/table-page/batch-actions-table.component.ts

```diff
@@ -1,51 +1,82 @@
+import { CommonModule } from '@angular/common';
 
 @Component({
   selector: 'app-batch-actions-table',
   template: `
     <h1>Batch Actions</h1>
-    <tds-table id="batchTable" enable-multiselect vertical-dividers="false" compact-design="false" enable-responsive="false">
+    <tds-table #table id="batchTable" multiselect vertical-dividers="false" compact-design="false" responsive="false">
       <tds-table-toolbar table-title="Batch action">
         <div slot="end">
-          <tds-button type="primary" size="sm" text="Download" (click)="download()"></tds-button>
+          <tds-button (click)="download()" type="primary" size="sm" text="Download"></tds-button>
         </div>
       </tds-table-toolbar>
-      <tds-table-header>
+      <tds-table-header multiselect (tdsSelectAll)="selectAll($event)" [allSelected]="allSelected">
         <tds-header-cell column-key="truck" column-title="Truck type"></tds-header-cell>
         <tds-header-cell column-key="driver" column-title="Driver name"></tds-header-cell>
         <tds-header-cell column-key="country" column-title="Country"></tds-header-cell>
         <tds-header-cell column-key="mileage" column-title="Mileage"></tds-header-cell>
       </tds-table-header>
-      <tds-table-body #tableBody></tds-table-body>
+      <tds-table-body>
+          <tds-table-body-row *ngFor="let row of tableData;" selected="{{ row.selected }}" (tdsSelect)="handleRowSelect(row.id, $event)">
+              <tds-body-cell cell-key="truck">
+                  {{ row.truck }}
+              </tds-body-cell>
+              <tds-body-cell cell-key="driver">
+                  {{ row.driver }}
+              </tds-body-cell>
+              <tds-body-cell cell-key="country">
+                  {{ row.country }}
+              </tds-body-cell>
+              <tds-body-cell cell-key="mileage">
+                  {{ row.mileage }}
+              </tds-body-cell>
+          </tds-table-body-row>
+      </tds-table-body>
     </tds-table>
     <ng-template modal-dr />
   `,
   styles: [``],
-  imports:[ModalDirective],
+  imports:[ModalDirective, CommonModule],
   standalone: true,
-  schemas: [CUSTOM_ELEMENTS_SCHEMA]
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
 })
-export class BatchActionsTableComponent implements AfterViewInit {
-  @ViewChild('tableBody', { static: false}) tableBody: ElementRef
+export class BatchActionsTableComponent {
   @ViewChild(ModalDirective, { static: true }) modalDirective!: ModalDirective
-  renderer2 = inject(Renderer2)
-  @Input() tableData: TableData[]
+  @ViewChild('table', { static: false}) tableRef: ElementRef
 
-  ngAfterViewInit(): void {
-    this.renderer2.setAttribute(this.tableBody.nativeElement, 'body-data', JSON.stringify(this.tableData))
+  tableData = exampleData;
+
+  allSelected = this.tableData.every((row) => row.selected)
+
+  async download(){
+    this.modalDirective.showModal(JSON.stringify(await this.tableRef.nativeElement.getSelectedRows()))
```

Comment:
I think we shouldn't provide a method that gets the selected rows from the DOM. 
The developer already has access to the selected rows in their data object.

Reply:
The main reason being that querying the DOM wouldnt work on a table with pagination, async loading of rows, etc

Reply:
@mJarsater Or is this not a problem since selected rows are most likely in the DOM?

I do think that we shouldn't use that method in the example though. 

Mind if i replace that part?

Reply:
To be honest im not sure thats the no1 priority for now, but if you feel like you have time go for it. Just remember that a cleanup PR is also gonna be merged and may or may not change a lot of this.

Reply:
You have a point 👍